### PR TITLE
Switch to PF4 Modal

### PR DESCRIPTION
**Pull request by rszwajko**
Depends on PR #1533 

  1. NavigationConfirmationModal
     - use titleIconVariant='warning' instead of alert
     - use "lead" class on first paragraph for consistency
  2. About modal
  3. VmConsoleInstructionsModal
  4. VmDetails related modals: Snapshots, Nics, Discs
  5. Replace PF3 MessageDialog: HotPlutConfirmationModal, NextRunChangeConfirmationModal, RestoreConfirmationModal, DeleteConfirmationModal, SessionActivityTracker

### CODE REVIEW

**Comment by rszwajko** [comment]
## Screenshots

### Shutdown VM 
shutdown_vm_pf4
shutdown_vm_pf3
### Remove VM
remove_vm_pf4
remove_vm_pf3
### About
PF4:
about_pf4
PF3:
about_pf3
### New snapshot
snapshot_pf4
snaphot_pf3
### Nics
nics_pf4
nics_pf3
### Discs
disk_pf4
disk_pf3
### Console instructions
console_instructions_pf4
console_instructions_pf3
### Save changes
save_changes_pf4
save_changes_pf3
### Navigation confirmation
PF4 (without alert)
nanigation_pf4
PF3 vs P3 modal with PF4 alert:
confirmation_drop_changes

**Comment by rszwajko** [comment]
## Modals based on PF3 MessageDialog

###  Hotplug
hotplug_pf4
hotplug_p3

### Nextrun
nextrun_pf4
nextrun_pf3

### Delete snapshot
deletesnapshot_pf4
deletesnapshot_pf3

### Restore snapshot
restoresnapshot_pf4
restoresnapshot_pf3

### Delete Nic/Disk (only PF4 versions)
deletedisk_pf4
deletenic_pf4

### Sesssion timeout

Variant with new header:
sessiontimeout_pf4_attention
Attempt to fit existing question into the header (unfortunately it's too long even in English):
sessiontimout_pf4_long

**Comment by ovirt-infra** [comment]
All tests passed

**Comment by ovirt-infra** [comment]
All tests passed

**Comment by ovirt-infra** [comment]
All tests passed

**Comment by rszwajko** [comment]
Added PF3-independent CSS for About dialog. The styling is based on existing PF3 CSS to provide similar user experience. Screenshot of updated design below:

image

File: src/components/SessionActivityTracker.js [hunk 1]
```diff
@@ -71,22 +70,22 @@ class SessionActivityTracker extends React.Component {
...
+        <ConfirmationModal
           show={this.state.counter > 0 && this.state.counter <= TIME_TO_DISPLAY_MODAL}
-          primaryAction={() => this.setState({ showTimeoutModal: false, counter: config.get('userSessionTimeoutInterval') })}
-          secondaryAction={onLogout}
-          onHide={onLogout}
-          primaryContent={<p className='lead'>{ msg.sessionExpired() }</p>}
-          secondaryContent={(
+          onClose={onLogout}
+          title={msg.attention()}
+          body={msg.sessionExpired() }
```
Comment (sjd78) [inline src/components/SessionActivityTracker.js:77]: :thinking: Double check against PF4 guidelines

File: src/components/VmDetails/cards/DetailsCard/NextRunChangeConfirmationModal.js [hunk 2]
```diff
@@ -1,39 +1,21 @@
...
       show={show}
-      onHide={onCancel}
+      onClose={onCancel}
       title={msg.nextRunConfirmTitle()}
-      icon={<Icon type='pf' name='warning-triangle-o' />}
 
-      primaryContent={<div className='lead'>{msg.nextRunConfirmContent()}</div>}
-      secondaryContent={<div>{msg.nextRunConfirmContentDetail()}</div>}
+      body={msg.nextRunConfirmContent()}
+      subContent={msg.nextRunConfirmContentDetail()}
```
Comment (sjd78) [inline src/components/VmDetails/cards/DetailsCard/NextRunChangeConfirmationModal.js:15]: :thinking: Double check against PF4 guidelines

Title: "Some changes require a restart"
Content: "Some configurations changes will not be able to take effect until the Virtual Machine is restarted.  A power cycle needs to take place to pick up this new configuration."

File: src/components/VmDetails/cards/DisksCard/DiskImageEditor.js [hunk 3]
```diff
@@ -321,214 +324,210 @@ class DiskImageEditor extends Component {
...
-
-            <Form
-              horizontal
-              onSubmit={e => { e.preventDefault() }}
-              id={`${idPrefix}-modal-form`}
+          isOpen={this.state.showModal}
+          onClose={this.close}
+          variant={ModalVariant.medium}
+          position='top'
+          title={createMode ? msg.createNewDisk() : msg.editDisk()}
```
Comment (sjd78) [inline src/components/VmDetails/cards/DisksCard/DiskImageEditor.js:332]: :thinking: Double check against PF4 guidelines

File: src/components/VmDetails/cards/SnapshotsCard/NewSnapshotModal.js [hunk 4]
```diff
@@ -78,46 +80,59 @@ class NewSnapshotModal extends Component {
...
+              <Alert
+                variant='info'
+                isInline
+                title={msg.details()}
+                style={{ 'margin-bottom': '10px' }}
+              >
+                {msg.snapshotInfo() }
+              </Alert>
+            </Col>
+            <FormGroup bsClass='form-group col-sm-12 required' validationState={this.state.emptyDescription ? 'error' : null}>
```
Comment (sjd78) [inline src/components/VmDetails/cards/SnapshotsCard/NewSnapshotModal.js:116]: May need to add some more cols to the labels so the "Save Memory" label doesn't wrap (in English at least)

File: src/components/VmDetails/cards/SnapshotsCard/RestoreConfirmationModal.js [hunk 5]
```diff
@@ -35,38 +35,35 @@ class RestoreConfirmationModal extends React.Component {
...
+                dangerouslySetInnerHTML={{
+                  __html: msg.areYouSureYouWantToRestoreSnapshot({ snapshotName: `"<strong>${minDescription}</strong>"` }),
+                }}
+              />
+              { snapshotsThatWillBeDeleted.size > 0 && (
+                <div id={`${id}-secondary`}>
+                  {msg.nextSnapshotsWillBeDeleted()}
+                  {snapshotsThatWillBeDeleted.map((s) => <div key={s.get('date')}>{s.get('description')}</div>)}
+                </div>
+              )}
```
Comment (sjd78) [inline src/components/VmDetails/cards/SnapshotsCard/RestoreConfirmationModal.js:63]: :thinking: Double check against PF4 guidelines

**Review by sjd78 — CHANGES_REQUESTED** [review]
For the editor modals (disk, nic at least), keep this in mind: https://www.patternfly.org/v4/components/modal#with-dropdown.  We want the drop downs to be able to be able to extend past the modal container instead of making the modal contents scroll.

The majority of the comments are around rewording the titles and content to better follow the modal design guidelines. More specifically with confirmation dialogs.  See: https://www.patternfly.org/v4/components/modal/design-guidelines#types-of-modals

**Comment by ovirt-infra** [comment]
All tests passed

File: src/components/NavigationConfirmationModal/index.js [hunk 6]
```diff
@@ -1,25 +1,31 @@
...
+          {msg.yes()}
+        </Button>,
+        <Button id={`${idPrefix}-button-no`} key="cancel" variant="link" onClick={onNo}>
+          {msg.no()}
+        </Button>,
+      ]}
+    >
+      <p className='lead'>{msg.unsavedChangesConfirmMessage()}</p>
+      <p>{additionalNote}</p>
     </Modal>
```
[the remark this answers is not in the document]
Comment (rszwajko) [reply src/components/NavigationConfirmationModal/index.js]: > Title: "Are you sure you want to drop your changes?"
>

Too long phrases get truncated which looks bad. I've used a shortened form of your proposal - check the screenshots below.

File: src/components/VmConsole/VmConsoleInstructionsModal.js [hunk 7]
```diff
@@ -23,41 +23,33 @@ class VmConsoleInstructionsModal extends React.Component {
...
-              <div>{ msg.remoteViewerAvailable() }</div>
-              <div className={style['console-detail-container']}>
-                <dl>
-                  <dt>RHEL, CentOS</dt><dd>sudo yum install virt-viewer</dd>
-                  <dt>Fedora</dt><dd>sudo dnf install virt-viewer</dd>
-                  <dt>Ubuntu, Debian</dt><dd>sudo apt-get install virt-viewer</dd>
-                  <dt>Windows</dt><dd>{ msg.downloadVirtManagerMSI() }</dd>
-                </dl>
-              </div>
+        <Modal
```
[the remark this answers is not in the document]
Comment (rszwajko) [reply src/components/VmConsole/VmConsoleInstructionsModal.js:31]: The dialog is not used after PR #1585 but was not removed yet. I'll add comment there to remember about this one.

File: src/components/VmDetails/cards/NicsCard/NicEditor.js [hunk 8]
```diff
@@ -172,119 +175,116 @@ class NicEditor extends Component {
 
         <Modal
           id={modalId}
-          show={this.state.showModal}
-          onHide={this.close}
+          isOpen={this.state.showModal}
+          onClose={this.close}
+          title={createMode ? msg.addNewNic() : msg.editNic()}
```
[the remark this answers is not in the document]
Comment (rszwajko) [reply src/components/VmDetails/cards/NicsCard/NicEditor.js:180]: Similar as create/edit disk dialog: this one look OK to me.

File: src/components/VmDetails/cards/SnapshotsCard/NewSnapshotModal.js (hunk 4 above)
Comment (rszwajko) [reply src/components/VmDetails/cards/SnapshotsCard/NewSnapshotModal.js:116]: Dialogs with rich content (i.e. forms) required more refactorings. This dialogs was fixed together in with VM Details screen (or later). In the most recent PR #1592  this looks OK.

**Comment by ovirt-infra** [comment]
All tests passed

File: src/components/VmDetails/cards/DisksCard/DiskImageEditor.js (hunk 3 above)
Comment (sjd78) [reply src/components/VmDetails/cards/DisksCard/DiskImageEditor.js:332]: I just tagged everything I saw.  This one is good.

File: src/components/VmDetails/cards/DetailsCard/HotPlugConfirmationModal.js [hunk 9]
```diff
@@ -1,37 +1,20 @@
...
   const { msg } = useContext(MsgContext)
   return (
-    <MessageDialog
+    <ConfirmationModal
       show={show}
-      onHide={onCancel}
-      title={msg.hotPlugConfirmTitle()}
-      icon={<Icon type='pf' name='warning-triangle-o' />}
+      onClose={onCancel}
+      title={msg.hotPlugConfirmContent()}
```
Comment (sjd78) [inline src/components/VmDetails/cards/DetailsCard/HotPlugConfirmationModal.js:12]: I'd add a question mark to this title:

"Apply changes now with Hot Plug?"

**Review by sjd78 — CHANGES_REQUESTED** [review]
One more ? to a title and everything looks good.

**Comment by ovirt-infra** [comment]
All tests passed

**Review by sjd78 — APPROVED** [review]
LGTM

**Pull request merged**